src/ng-select/ng-select.component.ts

```diff
@@ -510,6 +513,22 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
         dropdownPanel.style.width = selectRect.width + 'px';
     }
 
+    private _autoPositionDropdown() {
+        const selectRect = this.elementRef.nativeElement.getBoundingClientRect();
+        const scrollTop = document.documentElement.scrollTop || document.body.scrollTop;
+        const offsetTop = selectRect.top + window.pageYOffset;
+        const height = selectRect.height;
+        const dropdownHeight = this.dropdownPanel.nativeElement.getBoundingClientRect().height;
+
+        if (offsetTop + height + dropdownHeight > scrollTop + document.documentElement.clientHeight) {
+          this.renderer.addClass(this.dropdownPanel.nativeElement, 'top');
+          this.renderer.addClass(this.elementRef.nativeElement, 'top');
+        } else {
+          this.renderer.addClass(this.dropdownPanel.nativeElement, 'bottom');
+          this.renderer.addClass(this.elementRef.nativeElement, 'bottom');
+        }
+    }
```

Comment:
You also need to remove class if it was already added. Currently as I tested both `top` and `bottom` classes are applied after top -> bottom -> top changes. It would be good to reuse existing dropdownPosition binding so you don't need to manually manipulate classes. You can create new variable called `currentDropdownPosition` which by default would be equal to dropdownPosition input.
```ts
ngOnChanges(changes: SimpleChanges) {
        if (changes.dropdownPosition) {
              this.currentDropdownPosition = changes.dropdownPosition.currentValue;
        }
}
```
in html bindings now you can use `currentDropdownPosition` instead of `dropdownPosition` and on your autoposition logic you only need to update `currentDropdownPostinion` value.

```ts
if (offsetTop + height + dropdownHeight > scrollTop + document.documentElement.clientHeight) {
     this.currentDropdownPostinion = 'top';
} else {
    this.currentDropdownPostinion = 'bottom';
}
```

Reply:
@mnewmedia and tests are missing also 🐙